docs/DeveloperGuide.adoc

```diff
@@ -297,57 +297,97 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* * *` |curious student |view a module's details |check out if I want to take the module
+
+|`* * *` |busy student |manage deadlines |keep track of my assignments and tasks
+
+|`* * *` |organized student |manage my grades |know roughly how well I'm doing in the module
+
+|`* *` |new student |schedule my modules |plan how to approach my future focus areas
+
+|`* *` |graduating student |check what modules I need to complete |
+
+|`* *` |computing student |plan my focus area modules |choose a specialization
+
+|`* *` |competitive student |calculate my GPA |
+
+|`* *` |busy student |prioritize certain modules (and their deadlines) |manage my time better
+
+|`* *` |organized student |manage files related to a module |add and tag files
+
+|`* *` |student |catalog past modules |access the content conveniently
+
+|`*` |student |generate a sample module planner |plan my modules
+
+|`*` |hardworking student |access past exam papers |
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `ModuleBook` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: Find and Track a Module
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to find a module with incomplete keyword (e.g. 'CS2')
+2.  ModuleBook shows a list of modules matching the keyword (e.g. 'CS2030, CS2040, CS2100, CS2101, etc.')
+3.  User requests to track a module
+4.  ModuleBook tracks the module
 +
 Use case ends.
 
-*Extensions*
+[discrete]
+=== Use case: Add a deadline for a tracked module
 
-[none]
-* 2a. The list is empty.
+*MSS*
+
+1.  User requests to view the specific tracked module
+2.  ModuleBook displays the module details
+3.  User requests to add a deadline
+4.  ModuleBook adds the deadline
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+*Extensions*
+
+[none]
+* 2a. The module is not currently tracked
+[none]
+** 2a1. User requests to track the module
+** 2a2. The module is now tracked
+
 +
+Use case resumes at step 3
+
+[%hardbreaks]
 [none]
-** 3a1. AddressBook shows an error message.
+* 3a. ModuleBook requests for a description
+[none]
+** 3a1. User provides a description
+** 3a2. ModuleBook requests for a datetime
+** 3a3. User provides a datetime
+
 +
-Use case resumes at step 2.
+Use case resumes at step 4.
 
 _{More to be added}_
 
 [appendix]
 == Non Functional Requirements
```

Comment:
There are many more NFRs that can be included. For example,
- Should work on both 32-bit and 64-bit enviroments
- Should be able to manage up to 200 modules without noticeable sluggishness in performance
- Should work without installation
- Should work without internet access
...